docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,207 @@
+# Sparsity pattern and sparse matrix
+
+An important property of the finite element method is that it results in *sparse matrices*
+for the linear systems to be solved. On this page the topic of sparsity and sparse matrices
+are discussed.
+
+```@contents
+Pages = ["sparse_matrix.md"]
+Depth = 2:2
+```
+
+## Sparsity pattern
+
+The sparse structure of the linear system depends on many factors such as e.g. the weak
```

Comment:
I'm a bit torn here since the derivative is not always the definition of the matrix (but implicitly assumed in the text below for e.g. reference to Poisson's equation). Yet, I find it logical introduce the coupling via the derivative, e.g.

----------
In general, a finite element problem requires solving an equation system
```math
\boldsymbol{r}(\boldsymbol{a}) = 0
```
to find $\boldsymbol{a}$. For nonlinear problems when using the Newton-Raphson method, 
or for linear problems, a system matrix is constructed as 
```math
K_{ij} = \frac{\partial r_i}{\partial a_i}
```
Each entry, $r_i$, in $\boldsymbol{r}$, typically depend on only a few of the unknowns, $a_j$, such that many entries in $K_{ij}$ will be zero. This so-called sparse structure of the linear system, $K_{ij}$, depends on many factors such as e.g. the weak ...

----------

With the current text, it is for example not obvious that $K_{ij}$ can be zero while $K_{ji}$ is not. (I.e. which is nonzero "DoFs `i` and `j` couple"). 

(Not sure if it makes sense to discuss cases when using alternative techniques for defining the stiffness matrix via quasi-newton methods here, perhaps just mention that when discussing the `coupling` keyword?)

Reply:
This point also picked my attention. I would not even to down to nonlinear problems here. The point is really about linear operators, which either appear naturally as part of a PDE description (see e.g. Navier-Stokes example or Poisson problem or even linear elasticity) or in the linearization (e.g. second variation in energy-based elasticity). Hence the sparse matrix is not directly about linear systems, but about the representation of the discretization of an operator which allows e.g. efficient linear solves and fast simulations. So I have something into this direction in my head:
```suggestion
The sparse structure of linear operators in the context of the simulation of partial differential equations depends on many factors such as e.g. the weak
```
but it needs further refinement. Does this make sense for you guys?